neon_mq_connector/utils/client_utils.py

```diff
@@ -94,22 +100,31 @@ def handle_mq_response(channel: Channel, method, _, body):
         """
         api_output = b64_to_dict(body)
 
-        # The Messagebus connector generates a unique `message_id` for each
-        # response message. Check context for the original one; otherwise,
-        # check in output directly as some APIs emit responses without a unique
-        # message_id
+        # Backwards-compat. handles `context` in response for raw `Message`
+        # objects sent across the MQ bus
         api_output_msg_id = \
             api_output.get('context',
                            api_output).get('mq', api_output).get('message_id')
-        # TODO: One of these specs should be deprecated
         if api_output_msg_id != api_output.get('message_id'):
-            LOG.debug(f"Handling message_id from response context")
+            # TODO: `context.mq` handling should be deprecated
+            LOG.warning(f"Handling message_id from response context")
         if api_output_msg_id == message_id:
             LOG.debug(f'MQ output: {api_output}')
             channel.basic_ack(delivery_tag=method.delivery_tag)
-            channel.close()
-            response_data.update(api_output)
-            response_event.set()
+            if isinstance(api_output.get('_part'), int):
```

Comment:
Maybe we could try to achieve this behaviour using Streams? https://www.rabbitmq.com/docs/streams

Reply:
I looked into streams briefly and it seemed to me they were more oriented at sending either larger quantities of data or broadcasting to multiple recipients based on [RMQ documented use cases](https://www.rabbitmq.com/docs/streams#use-cases).

I opted for this method because it appears to be simpler to implement with fewer changes to our client helpers.